src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp

```diff
@@ -2613,3 +2619,73 @@ AUTH_INFO_TYPE KinesisVideoClientWrapper::authInfoTypeFromInt(UINT32 authInfoTyp
         default: return AUTH_INFO_UNDEFINED;
     }
 }
+
+VOID KinesisVideoClientWrapper::logPrintFunc(UINT32 level, PCHAR tag, PCHAR fmt, ...)
+{
+    JNIEnv *env;
+    BOOL detached = FALSE;
+    STATUS retStatus = STATUS_SUCCESS;
+    jstring jstrTag = NULL, jstrFmt = NULL, jstrBuffer = NULL;
+    CHAR buffer[MAX_LOG_MESSAGE_LENGTH];
+
+    CHECK(mJvm != NULL && mGlobalJniObjRef != NULL);
+
+    INT32 envState = mJvm->GetEnv((PVOID*) &env, JNI_VERSION_1_6);
+    if (envState == JNI_EDETACHED) {
+        if (mJvm->AttachCurrentThread((PVOID*) &env, NULL) != 0) {
+            goto CleanUp;
+        }
+        detached = TRUE;
+    }
+    
+    va_list list;
+    va_start(list, fmt);
+    vsnprintf(buffer, MAX_LOG_MESSAGE_LENGTH, fmt, list);
```

Comment:
looks like `vsnprintf` is not available on Windows we'll have to change this to not fail the windows build

Reply:
nevermind... was looking at something a decade old